gno.land/pkg/gnoland/app.go

```diff
@@ -25,52 +27,41 @@ import (
 	// Only goleveldb is supported for now.
 	_ "github.com/gnolang/gno/tm2/pkg/db/_tags"
 	_ "github.com/gnolang/gno/tm2/pkg/db/goleveldb"
-	"github.com/gnolang/gno/tm2/pkg/db/memdb"
 )
 
+// AppOptions contains the options to create the gno.land ABCI application.
 type AppOptions struct {
-	DB dbm.DB
-	// `gnoRootDir` should point to the local location of the gno repository.
-	// It serves as the gno equivalent of GOROOT.
-	GnoRootDir       string
-	GenesisTxHandler GenesisTxHandler
-	Logger           *slog.Logger
-	EventSwitch      events.EventSwitch
-	MaxCycles        int64
-	// Whether to cache the result of loading the standard libraries.
-	// This is useful if you have to start many nodes, like in testing.
-	// This disables loading existing packages; so it should only be used
-	// on a fresh database.
-	CacheStdlibLoad bool
+	DB                dbm.DB             // required
+	Logger            *slog.Logger       // defaults to log.NewNoopLogger()
+	EventSwitch       events.EventSwitch // defaults to events.NewEventSwitch()
+	MaxCycles         int64              // defaults to 0 (unlimited)
+	InitChainerConfig                    // options related to InitChainer
 }
 
-func NewAppOptions() *AppOptions {
-	return &AppOptions{
-		GenesisTxHandler: PanicOnFailingTxHandler,
-		Logger:           log.NewNoopLogger(),
-		DB:               memdb.NewMemDB(),
-		GnoRootDir:       gnoenv.RootDir(),
-		EventSwitch:      events.NilEventSwitch(),
+func (c *AppOptions) validate() error {
+	// Required fields
+	if c.DB == nil {
+		return fmt.Errorf("no db provided")
 	}
+	return nil
 }
 
-func (c *AppOptions) validate() error {
+func (c *AppOptions) setDefaults() {
```

Comment:
I prefer keeping a validation that fails so we expect more explicit initialization.

What you can do is provide a helper that returns a default `AppOptions`, like `DefaultDevelopmentOptions()`, so that it sounds not production-ready.